lib/AST/Macro.cpp

```diff
@@ -880,6 +880,108 @@ MacroRange MacroSubstitution::ReplacementChildren(void) const noexcept {
   }
 }
 
+// Returns the first and last fully substituted tokens from a given macro
+// substitution, if they exist.
+std::pair<std::optional<Token>, std::optional<Token>>
+GetFirstAndLastFullySubstitutedTokens(
+  MacroSubstitution const &substitution) noexcept {
+
+  // Get the fully substituted first token
+  std::optional<Token> first_tok = std::nullopt;
+  {
```

Comment:
What is this for? Ensuring destruction of something inside of the `{`?

Reply:
Honestly it was to make it easier for me to copy/paste code between the two blocks.

Reply:
The block are ultimately unnecessary if I reuse existing declarations so I'll remove them